packages/date-picker/src/DatePicker/DatePicker.tsx

```diff
@@ -149,71 +145,63 @@ export const DatePicker: React.FunctionComponent<DatePickerProps> = ({
+              [datePickerStyles.disabled]: isDisabled,
+            }
+          )}
           id={id}
-          labelText={labelText}
-          icon={dateStart}
-          value={
-            value ? value.toLocaleDateString("en-US", dateFormatOptions) : ""
-          }
           disabled={isDisabled}
-          inputRef={inputRef}
-          description={description}
-          onFocus={() => setIsOpen(true)}
+          ref={buttonRef}
+          onClick={handleOpenClose}
           onKeyDown={e => handleKeyDown(e)}
-          readOnly
           {...inputProps}
-        />
+        >
+          <div className={datePickerStyles.startIconAdornment}>
+            <Icon icon={dateStart} title="presentation" />
+          </div>
+          <span className={datePickerStyles.value}>
+            {value ? value.toLocaleDateString("en-US", dateFormatOptions) : ""}
+          </span>
+        </button>
       </div>
       {isOpen && (
-        <div
-          ref={setPopperElement}
-          style={styles.popper}
-          {...attributes.popper}
-          className={cx(datePickerStyles.calendar, {
-            classNameAndIHaveSpokenToDST,
-          })}
-          role="dialog"
-          aria-modal="true"
-          aria-label={
-            value ? `Change date, ${value.toLocaleDateString()}` : "Choose date"
-          }
-        >
-          <DayPicker
-            selectedDays={value}
+        <FocusLock onDeactivation={handleReturnFocus}>
+          <Calendar
+            setPopperElement={setPopperElement}
+            styles={styles}
+            attributes={attributes}
+            classNameAndIHaveSpokenToDST={classNameAndIHaveSpokenToDST}
+            value={value}
             initialMonth={initialMonth}
             firstDayOfWeek={firstDayOfWeek}
-            disabledDays={[
-              ...(disabledDates ? disabledDates : []),
-              {
-                daysOfWeek: disabledDaysOfWeek
-                  ? daysToNumbers(disabledDaysOfWeek)
-                  : [],
-              },
-              disabledRange && {
-                from: disabledRange.from,
-                to: disabledRange.to,
-              },
-              disabledBefore && {
-                before: disabledBefore,
-              },
-              disabledAfter && {
-                after: disabledAfter,
-              },
-              disabledBeforeAfter && {
-                before: disabledBeforeAfter.before,
-                after: disabledBeforeAfter.after,
-              },
-            ]}
-            onDayClick={handleOnDayChange}
-            navbarElement={getNavbar}
-            classNames={defaultDatePickerClasses}
-            onKeyDown={e => handleKeyDown(e)}
+            disabledDates={disabledDates}
+            disabledDaysOfWeek={disabledDaysOfWeek}
+            disabledRange={disabledRange}
+            disabledBefore={disabledBefore}
+            disabledAfter={disabledAfter}
+            disabledBeforeAfter={disabledBeforeAfter}
```

Comment:
If you wanted to remove the amount of props you have to send and receive, you could write a util that takes all of these and returns the array that you end up passing into DayPicker's `disabledDays` prop. Then this would just become a single `disabledDays` prop.